src/controllers/settings.js

```diff
@@ -138,13 +151,26 @@ async function resendEmail (req, res) {
     throw fluentError('user-verify-token-error')
   }
 
-  await sendVerificationEmail(emailId)
-
-  return res.json({
-    success: true,
-    status: 200,
-    message: 'Sent the verification email'
-  })
+  try {
```

Comment:
Is there a way to not dup the same code as line 93?

Reply:
Yeah this is what I was complaining about re: the way we're doing error responses right now... I'll figure out some way to do so.

Reply:
Is it worth it to wrap L154-173 into a function?

Reply:
Hm actually in the `addEmail` case this can't happen because the record won't be present, I'll try a couple different options and see which is the least bad.